### PR TITLE
Support custom texture atlas formats

### PR DESCRIPTION
**Pull request by rh101**
This change allows a developer to add support for any texture atlas format, with the default registered format being the normal PLIST.

To register a new format, simply create the implementation based on ISpriteSheetLoader, and then register it via:
`void registerSpriteSheetLoader(std::string formatId, std::unique_ptr<ISpriteSheetLoader> loader);`

For instance, the PLIST implementation is registered like this:
`registerSpriteSheetLoader("PLIST", std::unique_ptr<PlistSpriteSheetLoader>(new (std::nothrow) PlistSpriteSheetLoader()));`

### CODE REVIEW

**Comment by halx99** [comment]
Please fix the ci compiling errors

**Comment by rh101** [comment]
It should be working now.

**Comment by rh101** [comment]
FYI, there is no need to merge this in.  I only created the pull request in case the functionality would be useful to you or anyone else using this engine, and at the very least it shows how to support alternative texture atlas formats. :)

**Comment by aismann** [comment]
halx99: Why not add this PR to axde? Maybe with a documentation section too.
I think this is a good optional feature too.

**Comment by rh101** [comment]
The modifications should not affect any game code using ADXE, since the default sprite sheet format is still the PLIST. I've been using this code for months now, including implementing a custom sprite sheet format with it, and it's been working fine.  Multiple formats can be supported at the same time.

**Review by halx99 — APPROVED** [review]

**Comment by halx99** [comment]
> halx99: Why not add this PR to axde? Maybe with a documentation section too.
> I think this is a good optional feature too.

After method code style fixed, this pr could be merged

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 1]
```diff
@@ -51,70 +51,173 @@ NS_CC_BEGIN
...
+        // append .png
+        texturePath = texturePath.append(".png");
+
+        CCLOG("cocos2d: SpriteFrameCache: Trying to use file %s as texture", texturePath.c_str());
+    }
+    addSpriteFramesWithDictionary(dict, texturePath, filePath, cache);
 }
 
-bool SpriteFrameCache::init()
+void PlistSpriteSheetLoader::Load(const std::string& filePath, Texture2D* texture, SpriteFrameCache& cache)
```
[the remark this answers is not in the document]
Comment (rh101) [reply cocos/2d/CCSpriteFrameCache.cpp]: No worries, I'll update it now

**Comment by aismann** [comment]
@rh101:  Please, can you add an example on cpp-test too?

**Comment by rh101** [comment]
> 
> 
> @rh101: Please, can you add an example on cpp-test too?
> and a small docu (wiki?)?

Sure, I'll sort that out.

**Comment by rh101** [comment]
A simple test has been added in the SpriteFrameCacheTest section.  It will load a JSON array format sprite sheet, which is a common and simple format used by several game engines.  That implementation along with the PLIST loader implementation (class `PlistSpriteSheetLoader` ) show how to load all relevant fields in the sprite atlas.

**Pull request merged**